### PR TITLE
Add `arcade new` Improvements

### PR DESCRIPTION
**Pull request by EricGustin**
# PR Description
This PR is a part of the community contributed toolkits story.

* `arcade new` now uses jinja templates
* `arcade new` now creates a "cookiecutter" toolkit equipped with everything a community contributed toolkit needs to be easily tested, published to PyPi, etc. as its own Github repo
* I created the following toolkit with `arcade new`:
    - [PyPi](https://pypi.org/project/arcade-local-file-management/0.1.5/)
    - [Github](https://github.com/EricGustin/local_file_management/tree/0.1.5)

### CODE REVIEW

File: arcade/arcade/core/catalog.py [hunk 1]
```diff
@@ -541,7 +541,7 @@ def get_wire_type_info(_type: type) -> WireTypeInfo:
     # Special case: Enum can be enumerated on the wire
     elif issubclass(type_to_check, Enum):
         is_enum = True
-        enum_values = [e.value for e in type_to_check]  # type: ignore[union-attr]
```
Comment (EricGustin) [inline arcade/arcade/core/catalog.py:544]: unrelated cleanup

File: arcade/arcade/templates/{{ toolkit_name }}/poetry.toml
Comment (nbarbettini) [inline arcade/arcade/templates/{{ toolkit_name }}/poetry.toml:1]: Is this file necessary?

File: arcade/arcade/templates/{{ toolkit_name }}/.ruff.toml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    "B904", # Previously "TRY200"
+    # Depends function in arg string
+    "B008",
+    # raise from (cli specific)
+    "B904",
+    # long message exceptions
+    "TRY003",
+    # subprocess.Popen
+    "S603",
+]
```
Comment (nbarbettini) [inline arcade/arcade/templates/{{ toolkit_name }}/.ruff.toml:55]: Could we remove these ignores? (I have been meaning to remove it in our own project too)

However, double check that removing this doesn't cause a new toolkit to immediately fail the stricter linting rules.

If some of these ignores need to remain, then remove the `# TODO work to remove these` part only 😛

File: arcade/arcade/cli/new.py [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
 import typer
+from jinja2 import Environment, FileSystemLoader, select_autoescape
 from rich.console import Console
 
 console = Console()
 
 # Retrieve the installed version of arcade-ai
 try:
-    VERSION = get_version("arcade-ai")
+    ARCADE_VERSION = get_version("arcade-ai")
```
[the remark this answers is not in the document]
Comment (EricGustin) [reply arcade/arcade/cli/new.py:16]: The `ARCADE_VERSION` is used to construct the toolkit's `arcade-ai` dependency version in its `pyproject.toml`. So for example, if `ARCADE_VERSION` is `0.1.1`, then the toolkit's `pyproject.toml` will have the dependency of `arcade-ai = "0.1.*"`

(When you commented this was not getting passed through to the toolkit created by arcade new, but I updated)

File: arcade/arcade/templates/{{ toolkit_name }}/poetry.toml
Comment (EricGustin) [reply arcade/arcade/templates/{{ toolkit_name }}/poetry.toml:1]: Removed

File: arcade/arcade/templates/{{ toolkit_name }}/.ruff.toml (hunk 2 above)
Comment (EricGustin) [reply arcade/arcade/templates/{{ toolkit_name }}/.ruff.toml]: Removed with no issues with `make check` 😀

Comment (nbarbettini) [reply arcade/arcade/templates/{{ toolkit_name }}/.ruff.toml]: Woot

File: arcade/arcade/templates/{{ toolkit_name }}/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+authors = ["{{ toolkit_author_name }} <{{ toolkit_author_email }}>"]
+
+[tool.poetry.dependencies]
+python = "^3.10"
+arcade-ai = "{{ arcade_version }}"
+
+[tool.poetry.dev-dependencies]
+pytest = "^8.3.0"
+pytest-cov = "^4.0.0"
+mypy = "^1.5.1"
```
Comment (Spartee) [inline arcade/arcade/templates/{{ toolkit_name }}/pyproject.toml:15]: should the extras for fastapi and evals for arcadeai be included here? `arcade dev` and `arcade evals` won't work without them. We should think about how we want to work with this long term too.

File: arcade/pyproject.toml [hunk 5]
```diff
@@ -55,12 +55,14 @@ types-pytz = "^2024.1"
 types-python-dateutil = "^2.8.2"
 types-PyYAML = "^6.0.0"
 poetry-plugin-export = "^1.7.0"
+Jinja2 = ">=2.7,<4.0.0"
```
[the remark this answers is not in the document]
Comment (EricGustin) [reply arcade/pyproject.toml]: Moved to dependency from dev dependency

File: arcade/arcade/templates/{{ toolkit_name }}/Makefile [hunk 6]
```diff
@@ -0,0 +1,57 @@
+.PHONY: help
+
+help:
+	@echo "🛠️ local_file_management Commands:\n"
```
[the remark this answers is not in the document]
Comment (EricGustin) [reply arcade/arcade/templates/{{ toolkit_name }}/Makefile]: nice catch. updated

File: arcade/arcade/templates/{{ toolkit_name }}/tests/test_{{ toolkit_name }}.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import pytest
+from arcade.sdk.errors import ToolExecutionError
+from {{ package_name }}.tools.hello import say_hello
```
[the remark this answers is not in the document]
Comment (EricGustin) [reply arcade/arcade/templates/{{ toolkit_name }}/tests/test_{{ toolkit_name }}.py:3]: updated tool file name to `hello.py`

File: arcade/arcade/templates/{{ toolkit_name }}/pyproject.toml (hunk 4 above)
Comment (EricGustin) [reply arcade/arcade/templates/{{ toolkit_name }}/pyproject.toml:15]: I added `arcade-ai = {version = "{{ arcade_version }}", extras = ["fastapi", "evals"]}` to the toolkits dev dependencies. Lmk if this addresses your concern

Comment (nbarbettini) [reply arcade/arcade/templates/{{ toolkit_name }}/pyproject.toml:14]: Reviewer note: The extras were later removed after a discussion between Eric and I. The dependency on `fastapi` is an _Arcade CLI_ dependency, it should not be a dependency declared by this toolkit.

File: arcade/arcade/cli/new.py [hunk 8]
```diff
@@ -102,147 +101,28 @@ def create_new_toolkit(directory: str) -> None:
...
+    toolkit_author_name = ask_question("Github owner username?")
+    toolkit_author_email = ask_question("Author's email?")
+
+    context = {
+        "package_name": package_name,
+        "toolkit_name": toolkit_name,
+        "toolkit_description": toolkit_description,
+        "toolkit_author_name": toolkit_author_name,
+        "toolkit_author_email": toolkit_author_email,
+        "arcade_version": f"{ARCADE_VERSION.rsplit('.', 1)[0]}.*",
```
Comment (nbarbettini) [inline arcade/arcade/cli/new.py:114]: nit, but we are not validating this anywhere and just trusting that it has a `.`

File: arcade/arcade/templates/{{ toolkit_name }}/README.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+   - Go to Settings > Secrets and variables > Actions
+   - Click "New repository secret"
+   - Name your secret `PYPI_TOKEN`
+   - Paste your API Token into the Secret field
+
+### Creating a Release
+
+1. Navigate to your Github repository and click on Releases
+2. Create a new tag that corresponds to the version in your toolkit's `pyproject.toml` file
+   > Note: This will be 0.0.1 for your first release
```
Comment (nbarbettini) [inline arcade/arcade/templates/{{ toolkit_name }}/README.md:44]: It almost feels like these instructions belong somewhere else. This part is not really a README for the toolkit, but rather for the _one-time process_ of creating a toolkit.

**Review by nbarbettini — APPROVED** [review]
Ship it ⛴️

Added some non-blocking nits and comments.

**Comment by nbarbettini** [comment]
Confirmed that all of @Spartee's review notes were either addressed in this PR, or will be filed as follow-ups by @EricGustin
